app/forms/support_user_invite_form.rb

```diff
@@ -0,0 +1,51 @@
+class SupportUserInviteForm < ApplicationForm
+  FORM_PARAMS = %i[first_name last_name email].freeze
+
+  attr_accessor :email, :first_name, :last_name, :service
+
+  validate :validate_support_user
+  validates :service, presence: true
+
+  def invite!
+    save_support_user
+  end
+
+  def as_form_params
+    { "support_user" => slice(FORM_PARAMS) }
+  end
+
+  private
+
+  def support_user
+    @support_user ||=
+      begin
+        user = user_klass
+          .with_discarded
+          .discarded
```

Comment:
Do we need to call this line?

Reply:
Yes. Otherwise it finds Support Users who are not discarded and uses them. This shouldn't happen, as finding existing users should cause an error.

Reply:
Makes sense. Thanks for explaining!